src/Avalonia.Controls/Automation/Peers/ControlAutomationPeer.cs

```diff
@@ -221,7 +221,7 @@ protected override bool IsOffscreenCore()
                 IsOffscreenBehavior.FromClip => Owner.GetTransformedBounds() is not { } bounds ||
                     MathUtilities.IsZero(bounds.Clip.Width) ||
                     MathUtilities.IsZero(bounds.Clip.Height),
-                _ => !Owner.IsVisible,
+                _ => !Owner.IsEffectivelyVisible,
```

Comment:
cc @grokys

Reply:
Other backends handle this only in-so-far as they handle the hit testing in Avalonia code. This is in contrast to iOS which is the only target platform that performs the hit testing on the OS side. Therefore, I argue that this adaptation is necessary.